Sources/AppcuesKit/Presentation/Traits/Appcues/AppcuesBackdropKeyholeTrait.swift

```diff
@@ -97,15 +98,85 @@ internal class AppcuesBackdropKeyholeTrait: BackdropDecoratingTrait {
         let oldKeyholeBezierPath = oldShape.path(for: oldTargetRectangle)
         oldMaskPath.append(oldKeyholeBezierPath)
 
-        let maskLayer = CAShapeLayer()
-        maskLayer.fillRule = .evenOdd
-        maskLayer.path = newMaskPath.cgPath
+        let maskLayer: CALayer
+
+        if case let .circle(blurRadius: newBlurRadius) = newShape, newBlurRadius > 0 {
+            let newStartPoint = CGPoint(
+                x: newKeyholeBezierPath.bounds.midX,
+                y: newKeyholeBezierPath.bounds.midY)
+            let newEndPoint = CGPoint(
+                x: newKeyholeBezierPath.bounds.maxX + newBlurRadius,
+                y: newKeyholeBezierPath.bounds.maxY + newBlurRadius)
+
+            // Start the gradient at the edge of the keyhole path.
+            // Doesn't matter whether we use x or y since it's a circle.
+            // Cap at 0.995 so we don't collide with the
+            let newStartLocation: Double = min(
+                0.995,
+                (newEndPoint.x - newBlurRadius - newStartPoint.x) / (newEndPoint.x - newStartPoint.x)
```

Comment:
is there any chance of div/0 here?

Reply:
It would require the blur radius to be 0 and the width of the target to also be 0, but possible in theory, so I've added a check.